Fix zero-found-injections failures

spxiwh opened the pull request:
I saw some failures in plotting jobs in an IMBH testing run, which was due to there being 0 found injections in the 300 + 300 injection set. The code should be able to handle this. A few tweaks are needed to allow this.

Note that some codes declare `/usr/bin/python` as the executable. This should always be `/usr/bin/env python` but I only fixed this in the codes that I touched (lots of cases have this wrong, but the EGG installation prevents this from being a problem in most cases).

Note also the events.py had some weird bit of code to allow one (I think!) to specify floats as input to the newsnr/effective SNR routines. This caused failures if the inputs were empty arrays (or length one arrays in many cases), so I've now fixed this to return output in the same format as the input (ie. if you give floats as inputs you get floats back, but give arrays of length 1 as input and you get arrays of length 1 back).

tdent commented on bin/hdfcoinc/pycbc_page_coinc_snrchi:
```diff
@@ -133,7 +137,7 @@ ax.set_yscale('log')
     cb = pylab.colorbar()
     cb.set_label(coloring[args.colorbar_choice][1], size='large')
-except TypeError:
+except (TypeError, ZeroDivisionError):
```
Perhaps you need a comment here to say what causes the ZeroDivisionError and/or how it does so? 
Is ZeroDivisionError raised when there is exactly one found inj?

spxiwh replied:
I think this is different versions of numpy raising different errors (I couldn't get a TypeError). But the ZeroDivisionError is raised when there are no injection triggers (as the comment below states). I think the code is trying to find the upper and lower bounds of the [logarithmic] colorbar and is failing with no data.

tdent replied:
right .. do you think it would be possible to see what happens if there is exactly 1 found inj, since the other line that is checking for 'not enough' was 
`if len(coloring[args.colorbar_choice][0]) < 2:`

spxiwh replied:
I've verified that the different errors are caused by different numpy (or matplotlib or ...) versions.

For the case of exactly one trigger the code seems to run, although the colorbar may not be so useful. I've changed the code to check for len == 0 now though.

Also be aware that matplotlib's behaviour is dependent on the backend in use (normally 'Agg', but even that isn't identical across operating systems). The point is that there still may be other ways that 0 injections could cause failures in different environments/OSes. For now let's just play "whack-a-mole" with this kind of thing.

tdent commented on pycbc/events/events.py:
```diff
@@ -190,7 +190,8 @@ def newsnr(snr, reduced_x2, q=6., n=2.):
     newsnr[ind] *= ( 0.5 * (1. + reduced_x2[ind] ** (q/n)) ) ** (-1./q)
 
-    if len(newsnr) > 1:
+    # If input was floats, return a float. Otherwise return numpy array.
```
I would prefer to say 'if inputs were floats' or 'for float inputs'

spxiwh replied:
This has been changed as suggested.

spxiwh commented:
Okay, I've addressed @tdent 's comments, removed the Travis stuff, which is already on master now and rebased. I'm hoping Travis should run fine now and this is (IMO) ready to merge.
